lib/transaction_search.rb

```diff
@@ -41,9 +41,10 @@ def define_search_method(new_method_name, old_method_name)
         do_search(@search_fields)
         add_optimizations
         sort_and_paginate
+        email_raw_export(params[:to_email], account_transaction_report) if params[:to_email]
         respond_to do |format|
           format.html { render layout: @layout if @layout }
-          format.csv { render text: Reports::AccountTransactionsReport.new(@order_details, date_range_field: @date_range_field).to_csv }
+          format.csv { render text: account_transaction_report.to_csv }
```

Comment:
I think there should not be the ability to directly access the CSV version. The backgrounding should probably happen here within the `format.csv`. You can use `current_user.email` as a fallback if `params[:to_email]` doesn't exist.